executor/infoschema_reader.go

```diff
@@ -220,11 +218,10 @@ func (c *statsCache) get(ctx sessionctx.Context) (map[int64]uint64, map[tableHis
 		return nil, nil, err
 	}
 
-	c.mu.Lock()
-	c.loading = false
 	c.tableRows = tableRows
 	c.colLength = colLength
 	c.modifyTime = time.Now()
+	c.loading = true
```

Comment:
I donot think `c.loading` is useful anymore. Since the entire `get` function is locked.

Reply:
done.